test/inspector/inspector-helper.js

```diff
@@ -0,0 +1,406 @@
+      timeoutId && clearTimeout(timeoutId);
+      callback();
+    };
+    this.sendAll_(commands, () => {
+      timeoutId = setTimeout(() => {
+        let s = '';
+        for (const id in this.messages_) {
+          s += this.messages_[id] + '\n';
+        }
+        assert(false, s.substring(0, s.length - 1));
+      }, TIMEOUT);
+    });
+  });
+};
+
+TestSession.prototype.expectMessages = function(expects) {
+  if (!(expects instanceof Array)) expects = [ expects ];
+  let done = false;
+  let callback = null;
+
+  this.messagefilter_ = (message) => {
+    if (expects[0](message))
+      expects.shift();
+    if (!expects.length) {
+      done = true;
+      this.messagefilter_ = null;
+      callback && callback();
+    }
+  };
+  return this.enqueue((cb) => {
+    if (done) {
+      cb();
+    } else {
+      const timeoutId = setTimeout(
+          () => assert(false, 'Response was not received: ' +
+                       expects[0]),
+          TIMEOUT);
+      callback = () => {
+        clearTimeout(timeoutId);
+        cb();
+      };
+    }
+  });
+};
+
+TestSession.prototype.expectStderrOutput = function(regexp) {
+  const callbackPromise = new Promise((resolve) => {
+    this.enqueue((callback) => resolve(callback));
+  });
+  const timeoutId = setTimeout(() =>
+      assert(false, 'Timed out waiting for ' + regexp), TIMEOUT);
+  this.harness_.addStderrFilter(regexp, () => {
+    clearTimeout(timeoutId);
+    callbackPromise.then((callback) => callback());
+  });
+  return this;
+};
+
+TestSession.prototype.runNext_ = function() {
+  if (this.task_) {
+    setTimeout(() => {
+      this.task_(() => {
+        this.task_ = this.task_.next_;
+        this.runNext_();
+      });
+    });
+  }
+};
+
+TestSession.prototype.enqueue = function(task) {
+  if (!this.task_) {
+    this.task_ = task;
+    this.runNext_();
+  } else {
+    let t = this.task_;
+    while (t.next_)
+      t = t.next_;
+    t.next_ = task;
+  }
+  return this;
+};
+
+TestSession.prototype.disconnect = function(childDone) {
+  return this.enqueue((callback) => {
+    this.expectClose_ = true;
+    this.harness_.childInstanceDone =
+        this.harness_.childInstanceDone || childDone;
+    this.socket_.end();
+    callback();
+  });
+};
+
+const Harness = function(port, process) {
+  this.port = port;
+  this.mainScriptPath = mainScript;
+  this.stderrFilters_ = [];
+  this.process_ = process;
+  this.childInstanceDone = false;
+  this.returnCode_ = null;
+  this.running_ = true;
+
+  process.stderr.on('data', (data) => {
+    const pending = [];
+    const msg = data.toString();
+    console.log('[err]', msg);
+    for (const filter of this.stderrFilters_)
+      if (!filter(msg)) pending.push(msg);
+    this.stderrFilters_ = pending;
+  });
+  process.on('exit', (code, signal) => {
+    assert(this.childInstanceDone, 'Child instance died prematurely');
+    this.returnCode_ = code;
+    this.running_ = false;
```

Comment:
Maybe it's a good idea to use `close` instead of `exit` to be sure all the data from the stderr stream has been received.